### PR TITLE
[LETS-498] Block the commands that can start copylogdb/applylogdb

### PR DESCRIPTION
**Pull request by joohok**
http://jira.cubrid.org/browse/LETS-498

Purpose

In the new HA architecture, copylogdb and applylogdb are no longer required. 
Instead of using the 'cubrid hb start' command, 
there are other commands available that can be used to execute copylogdb and applylogdb.
 - cubrid hb replication start
 - cubrid hb copylogdb/applylogdb start
 - cub_admin copylogdb/applylogdb

And these commands are required to be blocked.

Implementation

- In the us_Command_map, which represents a set of commands for heartbeat, comment out SC_COPYLOGDB, SC_APPLYLOGDB, and REPLICATION
- In the ua_Utility_Map, which represents a set of commands for cub_admin, comment out COPYLOGDB, APPLYLOGDB

NOTE 
* applylogdb and copylogdb can be re-used later in the future when DR concept is discussed. So I decided to preserve the codes (functions and enum)
* To minimize changes, all the code lines related to copylogdb/applylogdb will be retained. They will be removed or marked as unused in a later issue.

### CODE REVIEW

**Review by hornetmj — APPROVED** [review]

File: msg/en_US/utils.msg [hunk 1]
```diff
@@ -241,9 +241,6 @@ usage: %2$s heartbeat <command> [args]\n\
 Available command:\n\
     start          [database-name]\n\
     stop           [-h <host-name>] [-i] [database-name]\n\
-    copylogdb      <start|stop> [-h <host-name>] database-name node-name\n\
```
Comment (wrlawodms) [inline msg/en_US/utils.msg:244]: I've found that other msgs are about these utilities with text search in the msg file. Don't they need to be removed?
1. cub_admin 
2. utility help msg for itself: copylogdb, applylogdb. Are they left since any path can't print them?

and applyinfo will be handled in another issue?

File: src/executables/util_service.c [hunk 2]
```diff
@@ -233,12 +235,14 @@ static UTIL_SERVICE_OPTION_MAP_T us_Command_map[] = {
   {ACCESS_CONTROL, COMMAND_TYPE_ACL, MASK_SERVER | MASK_BROKER | MASK_GATEWAY},
   {RESET, COMMAND_TYPE_RESET, MASK_BROKER | MASK_GATEWAY},
   {INFO, COMMAND_TYPE_INFO, MASK_BROKER | MASK_GATEWAY},
-  {SC_COPYLOGDB, COMMAND_TYPE_COPYLOGDB, MASK_HEARTBEAT},
-  {SC_APPLYLOGDB, COMMAND_TYPE_APPLYLOGDB, MASK_HEARTBEAT},
   {GET_SHARID, COMMAND_TYPE_GETID, MASK_BROKER},
   {TEST, COMMAND_TYPE_TEST, MASK_BROKER},
+#if defined (ENABLE_UNUSED_FUNCTION)
+  {SC_COPYLOGDB, COMMAND_TYPE_COPYLOGDB, MASK_HEARTBEAT},
```
Comment (wrlawodms) [inline src/executables/util_service.c:244]: Nit. COMMAND_TYPE_COPYLOGDB can be removed along. The same goes for the others.

File: src/executables/util_service.c [hunk 3]
```diff
@@ -88,11 +88,13 @@ typedef enum
   ACCESS_CONTROL,
   RESET,
   INFO,
-  SC_COPYLOGDB,
```
Comment (wrlawodms) [inline src/executables/util_service.c]: I'm wondering why some of enum or definition gets out, but some are not. 
If you're intending to minimize the change, I think you can leave this SC_COPYLOGDB/SC_APPLYLOGDB. It seems sufficient to modify us_Command_map.
 And if you're removing all related, you should remove others like COMMAND_TYPE_COPYLOGDB/COMMAND_TYPE_APPLYLOGDB, us_hb_process_copylogdb(), us_hb_process_applylogdb(), us_hb_copylogdb_start/stop(), UTIL_COPYLOGDB, etc..

File: src/executables/util_service.c [hunk 4]
```diff
@@ -5176,13 +5186,15 @@ process_heartbeat (int command_type, int argc, const char **argv)
     case RELOAD:
       status = process_heartbeat_reload (argc, argv);
       break;
+#if defined (ENABLE_UNUSED_FUNCTION)
```
Comment (wrlawodms) [inline src/executables/util_service.c]: I don't really know, but in this function, there is a path executing copylogdb/applylogdb. Are they blocked?

process_heartbeat() -> us_hb_process_start() -> us_hb_copylogdb/applylogdb_start()

**Review by wrlawodms — COMMENTED** [review]
It's hard to review all paths executing copylogdb/apploydb are blocked. I think they should be reviewed, right :)? Could you list the path which executed the processes to help us review it?

File: src/executables/util_service.c (hunk 4 above)
Comment (wrlawodms) [reply src/executables/util_service.c]: oh.. they are removed in #4364.

File: msg/en_US/utils.msg (hunk 1 above)
Comment (joohok) [reply msg/en_US/utils.msg:244]: Oh.. copylogdb and applylogdb can be executed using cub_admin.. Thanks!! I will change it.

1. I will block the copylogdb/applylogdb executed by cub_admin, and remove the 'copylogdb' and 'applylogdb' in the msg file
2. help msg for copylogdb and applylogdb will be left. because it canl be reused in the future. 
3. utilities are handled in another issue.

File: src/executables/util_service.c (hunk 3 above)
Comment (joohok) [reply src/executables/util_service.c]: OK, you are right 🥲
I will find and remove them all.

**Review by joohok — COMMENTED** [review]
> It's hard to review all paths executing copylogdb/apploydb are blocked. I think they should be reviewed, right :)? Could you list the path which executed the processes to help us review it?

I initially intended to remove only the part that receives commands (copylogdb, applylogdb), 
which can be done by removing commands in `us_Command_map`.

All other changes were made to remove unnecessary parts that became obsolete as copylogdb and applylogdb are no longer executed. 

However, thanks to your review, I've realized that I also need to prevent copylogdb and applylogdb from being executed through cub_admin.

I will provide the paths where copylogdb and applylogdb can be executed. Thank you.

File: src/executables/util_service.c [hunk 5]
```diff
@@ -1783,7 +1791,7 @@ process_server (int command_type, int argc, char **argv, bool show_usage, bool c
 		      if (status == NO_ERROR)
 			{
 			  (void) process_javasp (command_type, 1, (const char **) &token, false, true,
-					     process_window_service, false);
+						 process_window_service, false);
```
Comment (wrlawodms) [inline src/executables/util_service.c]: unrelated change?

Comment (joohok) [reply src/executables/util_service.c]: Yes, it is the existing code style error..

**Comment by hornetmj** [comment]
> I will provide the paths where copylogdb and applylogdb can be executed.

What do you mean in the previous comment. Do you mean leave it or remove it?

**Comment by joohok** [comment]
@hornetmj  
I think it would be better to only block the execution of copylogdb/applylogdb while keeping the related code intact. 
This approach aims to minimize changes. Removing (marking unused) unnecessary code can be done at a later stage.

The blocking should be implemented in the following two areas:

1. The part where the copylogdb/applylogdb commands are executed through `cub_admin`.
2. The part where the copylogdb/applylogdb/replication commands are executed through `cubrid hb`

Furthermore, I would like to remove help messages related to copylogdb/applylogdb/replication from the msg.

**Comment by hornetmj** [comment]
> @hornetmj I think it would be better to only block the execution of copylogdb/applylogdb while keeping the related code intact. This approach aims to minimize changes. Removing (marking unused) unnecessary code can be done at a later stage.
> 
> The blocking should be implemented in the following two areas:
> 
> 1. The part where the copylogdb/applylogdb commands are executed through `cub_admin`.
> 2. The part where the copylogdb/applylogdb/replication commands are executed through `cubrid hb`
> 
> Furthermore, I would like to remove help messages related to copylogdb/applylogdb/replication from the msg.

I understand what you mean(your approach).

**Comment by joohok** [comment]
Updated the PR description

File: src/executables/util_service.c [hunk 6]
```diff
@@ -233,12 +233,14 @@ static UTIL_SERVICE_OPTION_MAP_T us_Command_map[] = {
   {ACCESS_CONTROL, COMMAND_TYPE_ACL, MASK_SERVER | MASK_BROKER | MASK_GATEWAY},
   {RESET, COMMAND_TYPE_RESET, MASK_BROKER | MASK_GATEWAY},
   {INFO, COMMAND_TYPE_INFO, MASK_BROKER | MASK_GATEWAY},
-  {SC_COPYLOGDB, COMMAND_TYPE_COPYLOGDB, MASK_HEARTBEAT},
-  {SC_APPLYLOGDB, COMMAND_TYPE_APPLYLOGDB, MASK_HEARTBEAT},
   {GET_SHARID, COMMAND_TYPE_GETID, MASK_BROKER},
   {TEST, COMMAND_TYPE_TEST, MASK_BROKER},
+#if defined (ENABLE_UNUSED_FUNCTION)
```
Comment (wrlawodms) [inline src/executables/util_service.c]: How about just commenting it out with a short description instead of ENABLE_UNUSED_FUNCTION?
Since 
1. it's simply semantically wrong. It's not a function.
2. When setting on the ENABLE_UNUSED_FUNCTION, the copylogdb and applylogdb are enabled. It seems weird.

What do you think?

Comment (joohok) [reply src/executables/util_service.c]: Agree ! 

`ENABLE_UNUSED_FUNCTION`
Throughout the current CUBRID source code, we are using the above macro for parts that are currently not being used (enums in system_parameter.h, functions, or parts of functions...). 

However, I agree with your suggestion that the macro should only be used for functions that are truly unused based on their intended purpose (semantic). I will follow your suggestion.

**Review by wrlawodms — APPROVED** [review]

File: src/executables/util_service.c (hunk 2 above)
Comment (cristiarg) [reply src/executables/util_service.c:244]: Nit.
Shouldn't the the changes be consistent somehow? Like, the command types are removed but the commands in the command map are retained. Either guard them all with `#if 0` or remove them altogether.

**Review by cristiarg — APPROVED** [review]

File: src/executables/util_service.c (hunk 2 above)
Comment (joohok) [reply src/executables/util_service.c:244]: Agreed.
Other parts related to copylogdb/applylogdb will be guarded with `#if 0` in another issue (to minimize changes for now).

**Review by hornetmj — APPROVED** [review]
